### DISCUSSION TITLE
Bypassing lint/prettier errors

### DISCUSSION THREAD

**Discussion by eshdotcom** [UW-FLOM/flom-react#112] [Q&A] [answered]
Quick question: what is the best way to bypass the hundreds of `prettier` errors that we see on startup, as seen below? I have tried adding a `.prettierignore` file, but this did not change the output.

![prettier-errors](https://user-images.githubusercontent.com/65802389/144321007-ac376e25-a035-46b7-8a4a-a720db7e3287.PNG)

**Comment by VulpesVulpes825** [comment]
This is interesting. Are you on latest commit? In fact, prettier check is passed in the last few commits.

I checked [stack-overflow](https://stackoverflow.com/questions/53516594/why-do-i-keep-getting-delete-cr-prettier-prettier), it seems a windows specific issue. I will upload a change to eslint setting to fix the issue soon.

**Reply by VulpesVulpes825** [reply]
I have opened [pull request](https://github.com/UW-FLOM/flom-react/pull/113) for this, and will merge once CI passed. However, I strongly suggest next time if the CI passed all tests and build successfully, you might need to check if the issue is related to the operation system.

**Reply by VulpesVulpes825** [reply]
I have merged the pull request that should fix this issue. Pull the latest comment and run `yarn prettier-check` in the client folder to see if the issue is fixed.

**Reply by eshdotcom** [reply]
Thanks for merging the pull request. I pulled the latest commit and ran `yarn prettier-check` in `client`, however the issue persists. I ran `yarn dev:start` as well to ensure the images are up to date.

**Reply by VulpesVulpes825** [reply]
As I said in the initial reply, this is an issue that only appear on Windows. I do not have a Windows system on hand, so you might be on your own for this one. Considering moving what I just changed to the override section and see if the issue is fixed. There might be other answer in stackoverflow that might fix the issue.

**Reply by VulpesVulpes825** [reply]
Also, what is your editor? This error exist because you are using CRLF rather than LF. You need to change your code editor's setting for this.

**Comment by VulpesVulpes825** [comment, marked as answer]
Set your code editor to use LF as line ending, rather than CRLF used in Windows. Check your code editor's doc for how to change it. For Jetbrain and VSCode, it is on the bottom right.

**Reply by eshdotcom** [reply]
I've been having VSCode use LF as line ending, so the source of the issue must be a Windows system nuance, as you mentioned, or other issue. I'll continue looking into it.

**Reply by VulpesVulpes825** [reply]
I have added gitattributes file to  force EOL consistency.

Also try using `git config --global core.autocrlf input` to force using LF globally, or remove the flag for project config.

**Reply by VulpesVulpes825** [reply]
Hopefully commit cc359cc can fix this issue. Try pull the latest commit and actually run `yarn lint` in the client folder to see if eslint still cries about this.

**Reply by eshdotcom** [reply]
Even with the `.gitattributes` file from the latest commit, and having run `git config --global core.autocrlf input`, the error still persists. I ran `yarn lint` from the client directory, and it ran with no errors. I can't imagine anything else that needs to be changed, as stack-overflow and other sources all suggest each of the changes we have made.

**Reply by VulpesVulpes825** [reply]
As always, have you tried `yarn dev:destroy` and `docker system prune -a --volumes` to reset the development environment? 

You need to at least run `yarn dev:destroy` to destroy the docker when there is a change to any package.json or yarn.lock, which requires docker to rebuild node_modules. 

Also, please pull the latest commit, as I found out the yarn v3 upgrade creates more issues than expected. I have tested on my end with the latest patches. Docker should run without any issues.

**Reply by eshdotcom** [reply]
That did the trick, I'm able to see the application now. Thank you!
